Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3142,18 +2045,19 @@ private void setActivityLifeCycleObserver(Application application) {
 
         @Override
         public void onActivityCreated(Activity activity, Bundle bundle) {
+            intentState_ = INTENT_STATE.PENDING;
             if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
                 BranchViewHandler.getInstance().showPendingBranchView(activity);
             }
         }
 
         @Override
         public void onActivityStarted(Activity activity) {
+            intentState_ = INTENT_STATE.PENDING;
```

Comment:
Some times even on create is calling with null data on second link click.
So there seems no better way to figure out a stale intent. Will check
furrher and update the code if there is a unique way to find satle intent
always
On Jul 21, 2016 10:58 PM, "Alex Austin" <notifications@github.com> wrote:

> In Branch-SDK/src/io/branch/referral/Branch.java
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/271#discussion_r71830094>
> :
>
> >              if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
> >                  BranchViewHandler.getInstance().showPendingBranchView(activity);
> >              }
> >          }
> >
> >          @Override
> >          public void onActivityStarted(Activity activity) {
> > +            intentState_ = INTENT_STATE.PENDING;
>
> Doesn't this mean that we won't do any initialization until onResume is
> called now? I thought we discussed just doing this in the scenario where we
> determine the intent has stale data?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/271/files/5a7e8a3a10d3a3a67061a05f7a23c9124671d6ed#r71830094>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHTA9FL2Ht9rwWK1nIhk00foaKpqF3bpks5qYFwMgaJpZM4JSWfm>
> .
>